mars/services/scheduling/worker/execution.py

```diff
@@ -64,19 +64,25 @@ def __init__(self, subtask: Subtask,
         self._slot_manager_ref = slot_manager_ref
         self._slot_id = None
         self._enable_kill_slot = enable_kill_slot
+        self._slot_address = None
         self._should_kill_slot = False
 
     @property
     def slot_id(self):
         return self._slot_id
 
+    @property
+    def slot_address(self):
+        return self._slot_address
+
     def kill_slot_when_exit(self):
         if self._enable_kill_slot:  # pragma: no branch
             self._should_kill_slot = True
 
     async def __aenter__(self):
         self._slot_id = await self._slot_manager_ref.acquire_free_slot(
                 (self._subtask.session_id, self._subtask.subtask_id))
+        self._slot_address = await self._slot_manager_ref.get_slot_address(self.slot_id)
```

Comment:
If this line raises an exception, then the `self._slot_id` acquired will not be released.